src/stub/language/variable_name_options.rs

```diff
@@ -72,73 +65,75 @@ impl VariableNameOptions {
         }
     }
 
-    fn convert_to_snake_case(variable_name: &str) -> String {
-        SC_WORD_BREAK
-            .replace_all(variable_name, |caps: &regex::Captures| {
-                format!("{}_{}", &caps[1], &caps[2].to_lowercase())
+    fn ident_words(ident: &str) -> Vec<String> {
+        ident
+            .chars()
+            .peekable()
+            .batching(|char_iter| {
+                char_iter.peek()?; // check if there are any chars left
+
+                // The word boundary seem to be non-lowercase characters in CG
+                // Therefore we take
+                // boundary characters + lowercase characters until next boundary
+                let mut word_chars: Vec<char> =
+                    char_iter.peeking_take_while(|c| !c.is_ascii_lowercase()).collect();
+                word_chars.extend(char_iter.peeking_take_while(|c| c.is_ascii_lowercase()));
+
+                Some(word_chars.iter().collect::<String>().to_lowercase())
```

Comment:
arguably slightly more readable?
```suggestion
                Some(String::from_iter(word_chars).to_lowercase())
```

Reply:
Didn't know about this function, adding it.